frontend/pages/admin/IntegrationsPage/cards/AutomaticEnrollment/components/IdpSection/helpers.ts

```diff
@@ -0,0 +1,92 @@
+import { IEndUserAuthentication } from "interfaces/config";
+
+import validateUrl from "components/forms/validators/valid_url";
+
+export interface IFormDataIdp {
+  idp_name: string;
+  entity_id: string;
+  metadata_url: string;
+  metadata: string;
+}
+
+export const newFormDataIdp = (
+  config?: IEndUserAuthentication
+): IFormDataIdp => {
+  return {
+    idp_name: config?.idp_name || "",
+    entity_id: config?.entity_id || "",
+    metadata_url: config?.metadata_url || "",
+    metadata: config?.metadata || "",
+  };
+};
+
+const isEmptyFormData = (data: IFormDataIdp) => {
+  const values = Object.values(data);
+  return !values.length || values.every((v) => v === "");
+};
+
+const errorIdpName = (data: IFormDataIdp) => {
+  if (!data.idp_name) {
+    return "Identity provider name must be present.";
+  }
+  return "";
+};
+
+const errorEntityId = (data: IFormDataIdp) => {
+  if (!data.entity_id) {
+    return "Entity ID must be present.";
+  }
+  if (data.entity_id.length < 5) {
+    return "Entity ID must be 5 or more characters.";
+  }
+  return "";
+};
+
+const errorMetadataUrl = (data: IFormDataIdp) => {
+  if (!data.metadata && !data.metadata_url) {
+    return "Metadata or Metadata URL must be present.";
+  }
+  if (data.metadata_url && !validateUrl({ url: data.metadata_url })) {
+    return `${data.metadata_url} is not a valid URL.`;
+  }
+  return "";
+};
+
+const errorMetadata = (data: IFormDataIdp) => {
+  if (!data.metadata && !data.metadata_url) {
+    return "Metadata or Metadata URL must be present.";
+  }
+  return "";
+};
+
+const validators = {
```

Comment:
TODO: I noticed that the backend is not applying all of these validation rules. I'll follow up with a separate issue.